addons/account/models/account_move.py

```diff
@@ -2906,7 +2906,7 @@ def _post(self, soft=True):
             # environment.
             if (move.company_id.tax_lock_date and move.date <= move.company_id.tax_lock_date) and (move.line_ids.tax_ids or move.line_ids.tax_tag_ids):
                 move.date = move._get_accounting_date(move.invoice_date or move.date, True)
-                move.with_context(check_move_validity=False)._onchange_currency()
+            move.with_context(check_move_validity=False)._onchange_currency()
```

Comment:
I don't think this is a bug, but meant like this, as only in the case when it is earlier than the tax lock date you are supposed to change the date on the fly and as such change the currency.  @william-andre @jjscarafia

Reply:
Indeed, it shouldn't be needed to change this in the generic code; we don't want to recompute everything each time as it can needlessly override user input.

I don't know why it is needed, but it should be in the localization module, only when needed.

Reply:
hi @william-andre @jco-odoo 
one example of use case:
* create invoices from sale order and/or subscriptions on a Saturday
* leave them on draft
* on Monday a new rate is updated in odoo automatically
* post the invoices
* the rate of Saturday is used